app/src/main/java/org/mozilla/fenix/library/bookmarks/edit/EditBookmarkFragment.kt

```diff
@@ -175,8 +178,20 @@ class EditBookmarkFragment : Fragment() {
                     lifecycleScope.launch(IO) {
                         requireComponents.core.bookmarksStorage.deleteNode(guidToEdit)
                         requireComponents.analytics.metrics.track(Event.RemoveBookmark)
-                        launch(Main) {
+
+                        launch(Main) launchOnMain@{
                             Navigation.findNavController(requireActivity(), R.id.container).popBackStack()
+
+                            val rootView = activity.getRootView() ?: return@launchOnMain
```

Comment:
Could this just be an if statement or let call? Adding a label to return to is a bit unusual. Thanks! 😄

Reply:
It was a `let` you know, but it looked way too nested in terms of indentation and then i changed it! Will do this.

Reply:
Yeah there's no great way to use `let` with two objects ☹️ 

Sometimes I just use a if statement to check both variables, but I don't know if that works with the getters here.

Reply:
Yeah, i dont think it will work here, we'll have to capture the value returned by the function as it's mutable thus making its nullability in question, from the time it was checked to the time it was consumed.